docs/01_manual.md

```diff
@@ -0,0 +1,101 @@
+---
+title: Manual
+layout: home
+---
+
+Manual
+======
+
+TABLE OF CONTENTS
+-----------------
+
+- [SYNOPSIS](#synopsis)
+- [DESCRIPTION](#description)
+- [ARGUMENTS](#arguments)
+- [INSTALLATION AND CONFIGURATION](#installation-and-configuration)
+- [BUILDING FROM SOURCE](#building-from-source)
+
+
+SYNOPSIS
+--------
+
+    easy-deposit-agreement-generator [ -s ] <datasetID> <agreement-file>
+
+
+DESCRIPTION
+-----------
+
+A command line tool that creates a pdf document containing the deposit agreement for a given dataset. The tool searches for a dataset that corresponds 
+to the given `datasetID` and uses the metadata of this dataset, as well as the personal data of the depositor to generate the deposit agreement.
+
+**Note:** the dataset needs to be in Fedora already. Newly created datasets (for example from `easy-split-multi-deposit`) need to be ingested 
+by EASY first before generating the deposit agreement.
+
+`easy-deposit-agreement-generator` uses a template with placeholders. After replacing the placeholders with actual data, the template is converted into a PDF file.
+
+Placeholder substitution is achieved using [Apache Velocity](http://velocity.apache.org/), which fills in and merges a number of template HTML 
+files that are specified in `src/main/assembly/dist/res/template/`. Besides data from the dataset, several files in `src/main/assembly/dist/res/` 
+are required, namely `dans_logo.png`, `agreement_version.txt`, `Metadataterms.properties` and `velocity-engine.properties`.
+
+Pdf generation based on the assembled HTML is done using the command line tool [WeasyPrint](http://weasyprint.org/). Note that this tool 
+requires to be installed before being used by `easy-deposit-agreement-generator`. In order to not having this installed on our computers while developing 
+on this project or projects that depend on this project, we use an SSH connection to the development server where the command gets executed. 
+During development we therefore require a different script than the one that is used in production. Follow the steps below:
+
+1. In `application.properties` set `pdf.script=localrun.sh`;
+2. In `localrun.sh` fill in the variables `USER_HOST` and `PRIVATE_KEY`.
+
+A `--sample` or `-s` flag can be added to the command line tool to signal that a 'sample agreement' needs to be created. This version of the agreement
+can be created when the DOI is not yet calculated. Also in the title of the agreement it is clearly indicated that this version is a *sample*.
+
+ARGUMENTS
+---------
+
+     -s, --sample    Indicates whether or not a sample agreement needs to be created
+     -h, --help      Show help message
+     -v, --version   Show version of this program
+    
+    trailing arguments:
+     dataset-id (required)     The ID of the dataset of which a agreement has to be created
```

Comment:
This is also incorrect (copied from the old project). I would see a couple options:

* `easy-deposit-agreement-generator input-file output-file`: _gives paths to input and output file locations_
* `easy-deposit-agreement-generator -i input-file -o output-file`: _gives paths to input and output file locations, but with named arguments_
* `easy-deposit-agreement-generator input-file`: _gives path to input and writes output to stdout; this way you can do with the output whatever you want (a.k.a. `easy-deposit-agreement-generator file.json > output.pdf`)
* `easy-deposit-agreement-generator`: _gives no arguments, but reads from stdin and writes output to stdout; this way you can not only do with the output whatever you want, but also get the input from wherever (a.k.a. `cat file.json | easy-deposit-agreement-generator > output.pdf` or `curl ... | easy-deposit-agreement-generator > output.pdf`

I think all options should be supported. It's not difficult or much extra work to do so; I have done the same in `easy-check-user-references`.

Reply:
i think the cmi will not be used often, unlike `easy-check-user-references`, so a simple interface should suffice. I'm going for the second option, with named arguments, to avoid having to lookup the order every time you use this ...